tests/parametric/test_dynamic_configuration.py

```diff
@@ -517,3 +540,130 @@ def test_capability_tracing_custom_tags(self, library_env, test_agent, test_libr
         """Ensure the RC request contains the custom tags capability.
         """
         test_agent.wait_for_rc_capabilities([Capabilities.APM_TRACING_CUSTOM_TAGS])
+
+
+@scenarios.parametric
+@features.dynamic_configuration
+class TestDynamicConfigSamplingRules:
+    @parametrize("library_env", [{**DEFAULT_ENVVARS}])
+    def test_capability_tracing_sample_rules(self, library_env, test_agent, test_library):
+        """Ensure the RC request contains the trace sampling rules capability.
+        """
+        test_agent.wait_for_rc_capabilities([Capabilities.APM_TRACING_SAMPLE_RULES])
+
+    @parametrize(
+        "library_env",
+        [
+            {
+                **DEFAULT_ENVVARS,
+                "DD_TRACE_SAMPLING_RULES": json.dumps([{"sample_rate": ENV_SAMPLING_RULE_RATE, "service": "*"}]),
+            }
+        ],
+    )
+    def test_trace_sampling_rules_override_env(self, library_env, test_agent, test_library):
+        """The RC sampling rules should override the environment variable and decision maker is set appropriately.
+
+        When RC is unset, the environment variable should be used.
+        """
+        RC_SAMPLING_RULE_RATE_CUSTOMER = 0.8
+        RC_SAMPLING_RULE_RATE_DYNAMIC = 0.4
+        assert RC_SAMPLING_RULE_RATE_CUSTOMER != ENV_SAMPLING_RULE_RATE
+        assert RC_SAMPLING_RULE_RATE_DYNAMIC != ENV_SAMPLING_RULE_RATE
+        assert RC_SAMPLING_RULE_RATE_CUSTOMER != DEFAULT_SAMPLE_RATE
+        assert RC_SAMPLING_RULE_RATE_DYNAMIC != DEFAULT_SAMPLE_RATE
+
+        trace = get_sampled_trace(test_library, test_agent, service="", name="env_name")
+        assert_sampling_rate(trace, ENV_SAMPLING_RULE_RATE)
+        # Make sure `_dd.p.dm` is set to "-3" (i.e., local RULE_RATE)
+        span = trace[0]
+        assert "_dd.p.dm" in span["meta"]
+        # The "-" is a separating hyphen, not a minus sign.
+        assert span["meta"]["_dd.p.dm"] == "-3"
+
+        # Create a remote config entry with two rules at different sample rates.
+        set_and_wait_rc(
+            test_agent,
+            config_overrides={
+                "tracing_sampling_rules": [
+                    {
+                        "sample_rate": RC_SAMPLING_RULE_RATE_CUSTOMER,
+                        "service": TEST_SERVICE,
+                        "resource": "*",
+                        "provenance": "customer",
+                    },
+                    {
+                        "sample_rate": RC_SAMPLING_RULE_RATE_DYNAMIC,
+                        "service": "*",
+                        "resource": "*",
+                        "provenance": "dynamic",
+                    },
+                ]
+            },
+        )
```

Comment:
We should add tests with `"tags"` since that has a different format than local rules.